src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -66,7 +66,7 @@ internal void ReportAsyncParameterErrors(BindingDiagnosticBag diagnostics, Locat
                 {
                     diagnostics.Add(ErrorCode.ERR_BadAsyncArgType, getLocation(parameter, location));
                 }
-                else if (parameter.Type.IsUnsafe())
+                else if (parameter.Type.IsPointerOrFunctionPointer())
                 {
                     diagnostics.Add(ErrorCode.ERR_UnsafeAsyncArgType, getLocation(parameter, location));
```

Comment:
This question is relevant if we decide to adjust the wording of the message.

Reply:
The same applies here. `Task/ValueTask/MyTask<pointer>` is forbidden because of type argument restrictions. The message doesn't need to mention return types and arguably could be better if it didn't, but I don't plan to change it.